### PR TITLE
add github action to automate the release parts of the process

### PR DESCRIPTION
**Pull request by codeboten**
This PR adds a github action that can be triggered manually to do the following:

- creates tracking issue
- checks for blockers in core
- checks for blockers in contrib
- checks status of build-and-test in core
- checks status of build-and-test in contrib
- runs chlog update
- runs make-prepare for both stable/beta
- creates a PR

See the tracking issue created [here](https://github.com/codeboten/opentelemetry-collector/issues/175)
You can see the PR produced [here](https://github.com/codeboten/opentelemetry-collector/pull/176)

Running the workflow looks like this:
Screen Shot 2023-01-17 at 10 41 17 AM

Signed-off-by: Alex Boten <aboten@lightstep.com>

### CODE REVIEW

File: .github/workflows/prepare-release.yml
Comment (djaglowski) [inline .github/workflows/prepare-release.yml:16]:
```suggestion
        description: Release candidate version (beta)
```

File: .github/workflows/prepare-release.yml
Comment (MovieStoreGuy) [inline .github/workflows/prepare-release.yml:113]: Should this be done as part of this PR? Or is it fine to make it part of a future PR?

File: .github/workflows/prepare-release.yml
Comment (MovieStoreGuy) [inline .github/workflows/prepare-release.yml:132]: With the changes made to tooling, you shouldn't need to run `*install` anymore. Running the commands should check if the required command is built and build as required.

File: .github/workflows/prepare-release.yml
Comment (MovieStoreGuy) [inline .github/workflows/prepare-release.yml:150]: Since this is added in the release docs, does need to be documented here also?

**Review by MovieStoreGuy — COMMENTED** [review]
I have a few questions on this:

- What happens if someone chooses an existing version?
- Would it be possible to offer a dry run mode (similar to how terraform plan would do)

On my last question, it does feel a little wasteful and I am happy for it not to be included. I just really like the idea that releases have a reviewed process to help validate the release.

File: .github/workflows/prepare-release.yml
Comment (jpkrohling) [inline .github/workflows/prepare-release.yml:10]: Can you add an example, to make abundantly clear which kind of version this is about?

```suggestion
        description: Release candidate version (stable, like v1.0.0-rc4)
```

File: .github/workflows/prepare-release.yml
Comment (jpkrohling) [inline .github/workflows/prepare-release.yml:13]:
```suggestion
        description: Current version (stable, like v1.0.0-rc3)

```

File: .github/workflows/prepare-release.yml
Comment (jpkrohling) [inline .github/workflows/prepare-release.yml:16]:
```suggestion
        description: Release candidate version (beta, like v0.70.0)

```

File: .github/workflows/prepare-release.yml
Comment (jpkrohling) [inline .github/workflows/prepare-release.yml:19]:
```suggestion
        description: Current version (beta, like v0.69.1)
```

File: .github/workflows/prepare-release.yml
Comment (jpkrohling) [inline .github/workflows/prepare-release.yml:34]: How do you feel about having this in an actual source file?

File: .github/workflows/prepare-release.yml
Comment (jpkrohling) [inline .github/workflows/prepare-release.yml:33]: Do you want to have your repo here for the first version, or will you change before merging?

File: .github/workflows/prepare-release.yml
Comment (jpkrohling) [inline .github/workflows/prepare-release.yml:43]: The release manager would then still be required to create the individual issues, right?

File: .github/workflows/prepare-release.yml
Comment (jpkrohling) [inline .github/workflows/prepare-release.yml:60]: Should this happen even before creating the tracker issue?

File: .github/workflows/prepare-release.yml
Comment (jpkrohling) [inline .github/workflows/prepare-release.yml:113]: Or perhaps an approver's PAT as a secret?

File: .github/workflows/prepare-release.yml
Comment (jpkrohling) [inline .github/workflows/prepare-release.yml:150]: Is this pending because of technical challenges? Or because it just wasn't done yet?

File: .github/workflows/prepare-release.yml
Comment (codeboten) [reply .github/workflows/prepare-release.yml]: This has been removed.

File: .github/workflows/prepare-release.yml
Comment (codeboten) [reply .github/workflows/prepare-release.yml:37]: Unless we needed to re-use this code, I'm not sure if there's a value in having a separate script file, though I don't have a strong opinion on this.

File: .github/workflows/prepare-release.yml
Comment (codeboten) [reply .github/workflows/prepare-release.yml]: I've updated it

File: .github/workflows/prepare-release.yml
Comment (codeboten) [reply .github/workflows/prepare-release.yml:63]: I think it should be fine either ways, since the issue will only get created once.

File: .github/workflows/prepare-release.yml
Comment (codeboten) [reply .github/workflows/prepare-release.yml]: Removed

File: .github/workflows/prepare-release.yml
Comment (codeboten) [reply .github/workflows/prepare-release.yml]: I've removed it, the release note capture this step anyways, it can be automated later.

File: .github/workflows/prepare-release.yml
Comment (codeboten) [reply .github/workflows/prepare-release.yml:46]: That's correct. My hope is that through automation, the need for the tracking issue will be limited to an audit trail rather than a comprehensive reminder list :)

File: .github/workflows/prepare-release.yml
Comment (codeboten) [reply .github/workflows/prepare-release.yml]: It's possible, I've removed that step for now as it's still documented in the release doc as manual

File: .github/workflows/prepare-release.yml
Comment (jpkrohling) [reply .github/workflows/prepare-release.yml:37]: Being in a script means it could be used directly in case of necessity. It would also allow us to use a linter, like the shellcheck linter that is/was proposed as part of another PR.

**Review by jpkrohling — APPROVED** [review]

**Pull request merged**
